### PR TITLE
Porting the home account id logging to master.

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java [hunk 1]
```diff
@@ -132,6 +134,15 @@ public static String join(final char delimiter, @NonNull final Iterable<String>
                 && !StringExtensions.isNullOrBlank(uidUtidArray[INDEX_UTID])) {
             uid = uidUtidArray[INDEX_UID];
             utid = uidUtidArray[INDEX_UTID];
+        } else {
+            Logger.warnPII(TAG, "We had a home account id that could not be split correctly, " +
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java:138]:
```suggestion
            Logger.warn(TAG, "We had a home account id that could not be split correctly, " +
```
```suggestion
            Logger.warnPII(TAG, "We had a home account id that could not be split correctly, " +
```

File: common/versioning/version.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #Tue Apr 06 22:55:08 UTC 2021
-versionName=3.6.3
+versionName=3.6.3-debugAccountSplit
```
Comment (AdamBJohnsonx) [inline common/versioning/version.properties:2]:
```suggestion
versionName=3.6.5-RC1
```

**Comment by shahzaibj** [comment]
Why are we porting this to `master`?

**Comment by AdamBJohnsonx** [comment]
> Why are we porting this to `master`?

Because I need something to potentially release off of before the 4.0 train leaves?

**Comment by shahzaibj** [comment]
> > Why are we porting this to `master`?
> 
> Because I need something to potentially release off of before the 4.0 train leaves?

I didn't realize we were doing another release off of master here...are we doing it?

**Review by rpdome — APPROVED** [review]
Generally, if we're cutting a hotfix for specific issues, i would be more inclined to ship with that fix only.

It's just adding logger, so it's probably fine.

**Comment by rpdome** [comment]
> > > Why are we porting this to `master`?
> > 
> > 
> > Because I need something to potentially release off of before the 4.0 train leaves?
> 
> I didn't realize we were doing another release off of master here...are we doing it?

This is for unblocking OneAuth development.

**Pull request merged**